### PR TITLE
[Merged by Bors] - feat: port Order.Hom.Order

### PR DESCRIPTION
**Pull request by joneugster**

### CODE REVIEW

File: Mathlib/Order/Hom/Order.lean [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  by
+  constructor <;> intro h
+  · exact h 1 0
+  · intro n₁ n₂ a₁ a₂
+    have h' : ∀ n a₁ a₂, (f^[n]) (a₁ ⊔ a₂) ≤ (f^[n]) a₁ ⊔ a₂ := by
+      intro n
+      induction' n with n ih <;> intro a₁ a₂
+      · rfl
+      ·
+        calc
```
Comment (jcommelin) [inline Mathlib/Order/Hom/Order.lean:149]:
```suggestion
      · calc
```

File: Mathlib/Order/Hom/Order.lean [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+variable {α β : Type _}
+
+section Preorder
+
+variable [Preorder α]
+
+@[simps]
+instance [SemilatticeSup β] : HasSup (α →o β) where
+  sup f g :=⟨fun a => f a ⊔ g a, f.mono.sup g.mono⟩
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Hom/Order.lean:41]:
```suggestion
  sup f g := ⟨fun a => f a ⊔ g a, f.mono.sup g.mono⟩
```

File: Mathlib/Order/Hom/Order.lean [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+instance [SemilatticeSup β] : SemilatticeSup (α →o β) :=
+  { (_ : PartialOrder (α →o β)) with
+    sup := HasSup.sup
+    le_sup_left := fun _ _ _ => le_sup_left
+    le_sup_right := fun _ _ _ => le_sup_right
+    sup_le := fun _ _ _ h₀ h₁ x => sup_le (h₀ x) (h₁ x) }
+
+@[simps]
+instance [SemilatticeInf β] :
+    HasInf (α →o β) where inf f g := ⟨fun a => f a ⊓ g a, f.mono.inf g.mono⟩
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Hom/Order.lean:52]:
```suggestion
instance [SemilatticeInf β] : HasInf (α →o β) where
  inf f g := ⟨fun a => f a ⊓ g a, f.mono.inf g.mono⟩
```

File: Mathlib/Order/Hom/Order.lean [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  rfl
+#align order_hom.Inf_apply OrderHom.infₛ_apply
+
+theorem infᵢ_apply {ι : Sort _} [CompleteLattice β] (f : ι → α →o β) (x : α) :
+    (⨅ i, f i) x = ⨅ i, f i x :=
+  (infₛ_apply _ _).trans infᵢ_range
+#align order_hom.infᵢ_apply OrderHom.infᵢ_apply
+
+-- Porting note: This is a `syntactical tautology`, removed it for now.
+-- Can somebody confirm it's unecessary and remove it for good?
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Hom/Order.lean:92]: Would be good to have an answer here

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**